ldt/params/AquaCrop/AquaCrop_parmsMod.F90

```diff
@@ -0,0 +1,290 @@
+          units="m", &
+          full_name="Aquacrop compartment size")
+      Aquacrop_struc(n)%comp_size%vlevels = AquaCrop_struc(n)%max_comp
+      Aquacrop_struc(n)%comp_size%num_bins = AquaCrop_struc(n)%max_comp
+      allocate(Aquacrop_struc(n)%comp_size%value(&
+              LDT_rc%lnc(n),LDT_rc%lnr(n),&
+              Aquacrop_struc(n)%comp_size%vlevels))
+      call define_AC_compartments(n, AquaCrop_struc(n)%comp_size%value(:,:,:))
+
+
+    !! Read temperature climatology file
+    ! Read options from ldt.config
+    call ESMF_ConfigFindLabel(LDT_config,"AquaCrop temperature climatology directory:",rc=rc)
+    call ESMF_ConfigGetAttribute(LDT_config,AquaCrop_struc(n)%tempclimdir,rc=rc)
+    call LDT_verify(rc,"AquaCrop temperature climatology directory: not defined")
+
+    call ESMF_ConfigFindLabel(LDT_config,"AquaCrop reference year for climatology:",rc=rc)
+    call ESMF_ConfigGetAttribute(LDT_config,AquaCrop_struc(n)%tempcli_refyr,rc=rc)
+    call LDT_verify(rc,"AquaCrop reference year for climatology: not defined")
+
+    call ESMF_ConfigFindLabel(LDT_config,"AquaCrop temperature climatology spatial transform:",rc=rc)
+    call ESMF_ConfigGetAttribute(LDT_config,AquaCrop_struc(n)%tempclim_gridtransform,rc=rc)
+    call LDT_verify(rc,"AquaCrop temperature climatology spatial transform: not defined")
+
+    LDT_rc%monthlyData(n) = .true.
+
+    ! tmin
+        call set_param_attribs(Aquacrop_struc(n)%tmin_cli, "AC_Tmin_clim",&
+          units="K", &
+          full_name="minimum temperature climatology")
+      Aquacrop_struc(n)%tmin_cli%vlevels = 12
+      Aquacrop_struc(n)%tmin_cli%num_bins = 12
+      allocate(Aquacrop_struc(n)%tmin_cli%value(&
+              LDT_rc%lnc(n),LDT_rc%lnr(n),&
+              Aquacrop_struc(n)%tmin_cli%vlevels))
+
+    ! tmax
+        call set_param_attribs(Aquacrop_struc(n)%tmax_cli, "AC_Tmax_clim",&
+          units="K", &
+          full_name="maximum temperature climatology")
+      Aquacrop_struc(n)%tmax_cli%vlevels = 12
+      Aquacrop_struc(n)%tmax_cli%num_bins = 12
+      allocate(Aquacrop_struc(n)%tmax_cli%value(&
+              LDT_rc%lnc(n),LDT_rc%lnr(n),&
+              Aquacrop_struc(n)%tmax_cli%vlevels))
+
+      ! Call function to read monthly data
+      do k = 1,12
+        AquaCrop_struc(n)%tempclimfile = &
+            trim(AquaCrop_struc(n)%tempclimdir)//'tmin.'//&
+            trim(months(k))//'.txt'
+        call read_AC_Tclim(n, AquaCrop_struc(n)%tmin_cli%value(:,:,k))
+        AquaCrop_struc(n)%tempclimfile = &
+            trim(AquaCrop_struc(n)%tempclimdir)//'tmax.'//&
+            trim(months(k))//'.txt'
+        call read_AC_Tclim(n, AquaCrop_struc(n)%tmax_cli%value(:,:,k))
+      enddo ! end months
+  
+    enddo ! End nest
+    
+  end subroutine AquaCropParms_init
+
+ subroutine AquaCropParms_writeHeader(n,ftn,dimID,monthID)
+
+    use netcdf
+
+    integer   :: n,i 
+    integer   :: ftn
+    integer   :: dimID(3)
```

Comment:
I think this is the max number of dimensions I need which is 3: 2D space * 12 vertical levels (for soil compartments + for months for the climatology)

Reply:
You are right. It's not very clear why we would need dimID and ndimID. I think it has to do with the differences in the 3rd dimension.
This was copy-pasted from another model (CLM45) also having two different 3rd dimensions. In this case, the first 3rd dimension is the number of compartments (vertical levels, =12), the second 3rd dimension is the month number (12 months for the climatology). In this case, it's both times 12.
It could be simplified but I would be inclined to keep it if it works. But it's up to you.

As for the intent, it is never done in the other *_paramsMod.f90 units. So I would also keep it aligned with the rest.